### PR TITLE
chore: Upgrade typescript and tslint, fix new type issues

### PR DESCRIPTION
**Pull request by kimpers**
# Description

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.

### CODE REVIEW

File: src/utils/mesh_utils.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ import { orderUtils } from './order_utils';
 
 export const meshUtils = {
     orderInfosToApiOrders: (
-        orderEvent: Array<OrderEvent | AcceptedOrderInfo | RejectedOrderInfo | OrderInfo>,
+        orderEvent: (OrderEvent | AcceptedOrderInfo | RejectedOrderInfo | OrderInfo)[],
```
Comment (dekz) [inline src/utils/mesh_utils.ts:19]: I feel like VSCode always yelledd at me for this syntax over `Array<X>` 😂

File: test/utils/mesh_test_utils.ts [hunk 2]
```diff
@@ -87,13 +87,17 @@ export class MeshTestUtils {
         // NOTE(jalextowle): The way that Mesh validation currently works allows us
         // to only set the maker balance a single time. If this changes in the future,
         // this logic may need to be added to `addOrdersAsync`.
+        // tslint:disable-next-line:await-promise
```
Comment (dekz) [inline test/utils/mesh_test_utils.ts:90]: This isn't a promise?

**Review by dekz — APPROVED** [review]
Make it so!

File: src/utils/mesh_utils.ts (hunk 1 above)
Comment (kimpers) [reply src/utils/mesh_utils.ts:19]: Yeah they changed it to prefer `Something[]` in v4 🤷

File: test/utils/mesh_test_utils.ts (hunk 2 above)
Comment (kimpers) [reply test/utils/mesh_test_utils.ts:90]: TSLint is being silly, it's technically a `PromiseWithTransactionHash` 😂

**Comment by kimpers** [comment]
Closing this PR in favor of #365 which also contains these changes.

**Pull request closed without merging**